### PR TITLE
Presentation setup

### PR DESCRIPTION
**Pull request by pwyrwa**
@chrisauz9  
- Added some support classes and test case showing 4 ways to pass Function<> to method

### CODE REVIEW

File: src/main/java/com/dev9/presentation/model/DecoratedPerson.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    public String getName() {
+        return String.format("%s %s %s", decoration, person.getFirstName(), person.getLastName());
+    }
+
+
+    public static DecoratedPerson defaultDecorate(Person person) {
+        return new DecoratedPerson(person, DEFAULT_DECORATOR);
+    }
+
+    @Override public String toString() {
```
Comment (nimret) [inline src/main/java/com/dev9/presentation/model/DecoratedPerson.java]: I'd recommend using Apache Commons ToStringBuilder - less code :)

Comment (wiverson) [inline src/main/java/com/dev9/presentation/model/DecoratedPerson.java]: Guava FTW!  http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/MoreObjects.ToStringHelper.html

Comment (wiverson) [inline src/main/java/com/dev9/presentation/model/DecoratedPerson.java]: And I wrote a freakin' book on Apache Commons.

Comment (pwyrwa) [inline src/main/java/com/dev9/presentation/model/DecoratedPerson.java]: How about intellij cmd+n, generate toString ?

Comment (nimret) [inline src/main/java/com/dev9/presentation/model/DecoratedPerson.java]: all good choices, except generating the code inline imo

Comment (fastnsilver) [inline src/main/java/com/dev9/presentation/model/DecoratedPerson.java]: Lombok `@ToString` ? http://projectlombok.org/features/ToString.html

**Pull request merged**